### ISSUE TITLE
Bug: interrupt command execution

### ISSUE THREAD

**Issue by marcbelmont** [ekkinox/yai#34]
1. Write a prompt and type enter
2. While the response is streaming, press Ctrl+C
3. Expected behavior: the response stops streaming, and you go back to the yai prompt. Currently, the program exits.

**Comment by ekkinox** [comment]
You mean interrupting the chat mode? 

If yes, will handle this.

**Comment by marcbelmont** [comment]
Yes, it would be great to interrupt the response in chat mode without exiting the program.
